cli/deno_dir.rs

```diff
@@ -108,88 +152,70 @@ impl DenoDir {
 
     let root: PathBuf = custom_root.unwrap_or(default);
     let gen = root.as_path().join("gen");
+    let gen_cache = DiskCache::new(&gen);
     let deps = root.as_path().join("deps");
-    let deps_http = deps.join("http");
-    let deps_https = deps.join("https");
-
-    // Internally within DenoDir, we use the config as part of the hash to
-    // determine if a file has been transpiled with the same configuration, but
-    // we have borrowed the `State` configuration, which we want to either clone
-    // or create an empty `Vec` which we will use in our hash function.
-    let config = match state_config {
-      Some(config) => config.clone(),
-      _ => b"".to_vec(),
-    };
+    let deps_cache = DiskCache::new(&deps);
 
     let deno_dir = Self {
       root,
-      gen,
-      deps,
-      deps_http,
-      deps_https,
-      config,
+      gen_cache,
+      deps_cache,
       progress,
-      download_cache: DownloadCache::default(),
+      source_file_cache: SourceFileCache::default(),
+      use_disk_cache,
+      no_remote_fetch,
     };
 
     // TODO Lazily create these directories.
-    deno_fs::mkdir(deno_dir.gen.as_ref(), 0o755, true)?;
-    deno_fs::mkdir(deno_dir.deps.as_ref(), 0o755, true)?;
-    deno_fs::mkdir(deno_dir.deps_http.as_ref(), 0o755, true)?;
-    deno_fs::mkdir(deno_dir.deps_https.as_ref(), 0o755, true)?;
+    // TODO: once saving and loading of SourceFiles uses DiskCache API these calls can be removed
```

Comment:
This is already handled, so those calls can be removed - check if tests won't fail